### PR TITLE
add azure monitor log profile module

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

add new module azure_rm_monitorlogprofile. mark as WIP to avoid be merged into 2.8.

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yungezz** [comment]
HI @mattclay  seems test account doesn't have permission to create log profile, pls see  below error:
```
"Error creating/updating log profile: Azure Error: AuthorizationFailed\nMessage: The client 'xxxxxxxxxxxxxxxxxxx' with object id 'xxxxxxxxxxx' does not have authorization to perform action 'microsoft.insights/logprofiles/write' over scope '/subscriptions/xxxxxxxxxxxxxxxxxxxx'."
```
https://app.shippable.com/github/ansible/ansible/runs/117172/114/tests

Could you pls grant necessary permission? Thanks.

**Comment by mattclay** [comment]
@yungezz I don't think we'll be able to test this with our current CI system. This feature appears to operate at the subscription level, rather than the resource group or resource level. We don't have a safe way to expose privileges for managing the subscription currently.

You can still create tests and run them locally, but you'll need to mark the test `unsupported` in the aliases file to have it skipped in CI.

**Comment by yungezz** [comment]
got it. thanks Matt @mattclay .

**Comment by yungezz** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@yuwzho @zikalino review_need

**Comment by Fred-sun** [comment]
@yuwzho @zikalino Please help review this PR when you're available! Thanks a lot!

**Comment by Fred-sun** [comment]
@yuwzho @zikalino @nitzmahone Please take a review this when you're free? Thank you very much!

**Comment by Fred-sun** [comment]
@yuwzho @zikalino @nitzmahone  review_need

**Comment by yungezz** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@nitzmahone @jborean93 Please take a review this PR when you're free! Thank you very much!

File: lib/ansible/modules/cloud/azure/azure_rm_monitorlogprofile.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+    location:
+        description:
+            - Resource location.
+        type: str
+    locations:
+        description: List of regions for which Activity Log events should be stored.
+        type: list
+    categories:
+        description:
+            - List of categories of logs. These categories are created as is convinient to  user. Some Values are Write, Delete and/or Action.
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_monitorlogprofile.py]: convenient

File: lib/ansible/modules/cloud/azure/azure_rm_monitorlogprofile.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+            enabled:
+                description: Whether the retention policy is enabled.
+                type: bool
+            days:
+                description: The number of days for the retention. A value of 0 will retain the events indefinitely.
+                type: int
+    service_bus_rule_id:
+        description:
+            - The service bus rule  ID of  the service bus namespace in which you would like to have Event Hubs created for streaming in the Activity Log.
+            - format like {serice_bus_resource_id}/authorizationrules{key_name}.
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_monitorlogprofile.py]: service_bus_resource_id

**Review by nitzmahone — CHANGES_REQUESTED** [review]
couple little docs things, otherwise LGTM

**Comment by Fred-sun** [comment]
@nitzmahone It has been updated, please help to review it again!  thank you very much!

**Comment by zikalino** [comment]
all minor comments addressed now -- merging

**Pull request merged**